pallets/rolldown/src/lib.rs

```diff
@@ -311,6 +311,50 @@ pub mod pallet {
 			Ok(().into())
 		}
 
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::DbWeight::get().reads_writes(1, 1).saturating_add(Weight::from_parts(40_000_000, 0)))]
+		pub fn withdraw(
```

Comment:
please add unit tests for the method that covers basic scenarios:
- check if total allocation of tokens has changed 
- counter is updated

Reply:
done